sopel/tools/jobs.py

```diff
@@ -1,100 +1,79 @@
-    def __init__(self, lock):
-        self.lock = lock
-
-    def __enter__(self):
-        self.lock.release()
-
-    def __exit__(self, _type, _value, _traceback):
-        self.lock.acquire()
-
-
-class PriorityQueue(Queue.PriorityQueue):
-    """A priority queue with a peek method."""
-    def peek(self):
-        """Return a copy of the first element without removing it."""
-        self.not_empty.acquire()
-        try:
-            while not self._qsize():
-                self.not_empty.wait()
-            # Return a copy to avoid corrupting the heap. This is important
-            # for thread safety if the object is mutable.
-            return copy.deepcopy(self.queue[0])
-        finally:
-            self.not_empty.release()
+py3 = sys.version_info.major >= 3
 
 
 class JobScheduler(threading.Thread):
-
     """Calls jobs assigned to it in steady intervals.
 
     JobScheduler is a thread that keeps track of Jobs and calls them every
-    X seconds, where X is a property of the Job. It maintains jobs in a
-    priority queue, where the next job to be called is always the first
-    item.
-    Thread safety is maintained with a mutex that is released during long
-    operations, so methods add_job and clear_jobs can be safely called from
-    the main thread.
-
-    """
+    X seconds, where X is a property of the Job.
 
-    min_reaction_time = 30.0  # seconds
-    """How often should scheduler checks for changes in the job list."""
+    Thread safety is ensured with an internal mutex.
 
+    It runs forever until the :attr:`stopping` event is set using the
+    :meth:`stop` method.
+    """
     def __init__(self, bot):
         """Requires bot as argument for logging."""
         threading.Thread.__init__(self)
         self.bot = bot
-        self._jobs = PriorityQueue()
-        # While PriorityQueue it self is thread safe, this mutex is needed
-        # to stop old jobs being put into new queue after clearing the
-        # queue.
+        self.stopping = threading.Event()
+        self._jobs = []
         self._mutex = threading.Lock()
-        # self.cleared is used for more fine grained locking.
-        self._cleared = False
 
     def add_job(self, job):
         """Add a Job to the current job queue."""
-        self._jobs.put(job)
+        with self._mutex:
+            self._jobs.append(job)
 
     def clear_jobs(self):
         """Clear current Job queue and start fresh."""
-        if self._jobs.empty():
-            # Guards against getting stuck waiting for self._mutex when
-            # thread is waiting for self._jobs to not be empty.
-            return
         with self._mutex:
-            self._cleared = True
-            self._jobs = PriorityQueue()
+            self._jobs = []
+
+    def stop(self):
+        """Ask the job scheduler to stop.
+
+        The scheduler thread will stop its loop over jobs to process, but it
+        won't join the thread, or clear its queue—this has to be done
```

Comment:
Apropos your comment about dashes in #1402, this is fine as-is without spaces because it's a docstring. :)